runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcWindmillStreamFactory.java

```diff
@@ -260,6 +238,40 @@ public interface Builder {
 
     Builder setMaxBackOffSupplier(Supplier<Duration> maxBackOff);
 
+    Builder setSendKeyedGetDataRequests(boolean sendKeyedGetDataRequests);
+
+    Builder setProcessHeartbeatResponses(
+        Consumer<List<ComputationHeartbeatResponse>> processHeartbeatResponses);
+
+    /** Creates stream factory without health checking. */
     GrpcWindmillStreamFactory build();
+
+    /**
+     * Creates stream factory that sends health check RPCs every healthCheckIntervalMillis.
+     *
+     * @implNote Health checks are run on background daemon thread, which will be cleaned up when
```

Comment:
maybe expand comment

will *only* be cleaned up on JVM shutdown. Factories should be long-lived to avoid leaking memory.

Reply:
done